matRad_calcParticleDose.m

```diff
@@ -34,40 +34,54 @@
 % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
 if exist('param','var')
-   if ~isfield(param,'logLevel')
-      param.logLevel = 1;
-   end
-   % default: dose influence matrix computation
-   if ~isfield(param,'calcDoseDirect')
-      param.calcDoseDirect = false;
-   end
+    if ~isfield(param,'logLevel')
+        param.logLevel = 1;
+    end
+    % default: dose influence matrix computation
+    if ~isfield(param,'calcDoseDirect')
+        param.calcDoseDirect = false;
+    end
 else
-   param.calcDoseDirect = false;
-   param.subIx          = [];
-   param.logLevel       = 1;
+    param.calcDoseDirect = false;
+    param.subIx          = [];
+    param.logLevel       = 1;
+end
+
+% load machine file
+fileName = [pln.radiationMode '_' pln.machine];
+try
+    load([fileparts(mfilename('fullpath')) filesep fileName]);
+catch
+    matRad_dispToConsole(['Could not find the following machine file: ' fileName ],param,'error');
+end
+
+pln = matRad_checkHeterogeneity(pln,cst,machine.data(1));
```

Comment:
Does this function also work if the heterogeneity correction is turned off ?

Reply:
yes, the matRad_checkHeterogeneity function should select the required parameters only if the correction enabled, in all other cases, calcParticleDose works without restrictions.